TESTS/mbed_platform/Stream/main.cpp

```diff
@@ -18,43 +18,198 @@
 #include "utest/utest.h"
 #include "unity/unity.h"
 #include "mbed.h"
+#include "Stream.h"
+
+/* This test suite verifies that write/read/write/read sequence can be
+ * successfully executed on the Stream objects.
+ *
+ * A qute from `man 3 fdopen`:
+ *
+ * Reads and writes may be intermixed on read/write streams in any order. Note
+ * that ANSI C requires that a file positioning function intervene between
```

Comment:
Still don't like this man page text, as it's paraphrasing what the standard actually says, inaccurately.

If you're going to put anything in, make it the standard text. From C99:

> When a file is opened with update mode ('+' as the second or third character in the
> above list of mode argument values), both input and output may be performed on the
> associated stream. However, output shall not be directly followed by input without an
> intervening call to the fflush function or to a file positioning function (fseek,
> fsetpos, or rewind), and input shall not be directly followed by output without an
> intervening call to a file positioning function, unless the input operation encounters end-of-
> file.

Reply:
Updated the spec quote as suggested.